path_trie.go

```diff
@@ -14,13 +14,30 @@ type PathTrie struct {
 	children  map[string]*PathTrie
 }
 
+// PathTrieConfig for building a path trie with different segmenter
+type PathTrieConfig struct {
+	Segmenter StringSegmenter
+}
+
 // NewPathTrie allocates and returns a new *PathTrie.
 func NewPathTrie() *PathTrie {
 	return &PathTrie{
 		segmenter: PathSegmenter,
 	}
 }
 
+// NewPathTrieWithConfig allocates and returns a new *PathTrie with the given *PathTrieConfig
+func NewPathTrieWithConfig(config *PathTrieConfig) *PathTrie {
```

Comment:
I appreciate you're trying to avoid changing the existing signature. In https://github.com/dghubble/trie/pull/14 I was commenting that I'm actually fine with this being *the* NewPathTrie. This package has no releases and no reasonable expectation that exported functions won't be changed when there is a clear improvement (allowing the segmenter to be customized, which was always the intent)

Reply:
I'll go with it for now, though these may still get renamed at some point

Reply:
pls. check pull request #17